xdsl/dialects/irdl/irdl.py

```diff
@@ -35,12 +38,47 @@
     SymbolTable,
 )
 from xdsl.utils.exceptions import VerifyException
+from xdsl.utils.str_enum import StrEnum
 
 ################################################################################
 # Dialect, Operation, and Attribute definitions                                #
 ################################################################################
 
 
+class VariadicityEnum(StrEnum):
+    Single = "single"
+    Optional = "optional"
+    Variadic = "variadic"
```

Comment:
```suggestion
    SINGLE = "single"
    OPTIONAL = "optional"
    VARIADIC = "variadic"
```
It's ugly but standard :)